server/server_test.go

```diff
@@ -516,3 +517,164 @@ type okHTTPHandler struct{}
+							LoadBalancer: &types.LoadBalancer{Method: "Wrr"},
+						},
+					},
+				}
+			},
+			wantStatusCode: http.StatusOK,
+		},
+		{
+			desc: "No Frontend",
+			dynamicConfig: func(testServerURL string) *types.Configuration {
+				return &types.Configuration{
+					Frontends: map[string]*types.Frontend{},
+					Backends:  map[string]*types.Backend{},
+				}
+			},
+			wantStatusCode: http.StatusNotFound,
+		},
+		{
+			desc: "Empty Backend LB-Drr",
+			dynamicConfig: func(testServerURL string) *types.Configuration {
+				return &types.Configuration{
+					Frontends: map[string]*types.Frontend{
+						"frontend": {
+							EntryPoints: []string{"http"},
+							Backend:     "backend",
+							Routes:      map[string]types.Route{requestPath: {Rule: "Path:" + requestPath}},
+						},
+					},
+					Backends: map[string]*types.Backend{
+						"backend": {
+							Servers:      map[string]types.Server{},
+							LoadBalancer: &types.LoadBalancer{Method: "Drr"},
+						},
+					},
+				}
+			},
+			wantStatusCode: http.StatusServiceUnavailable,
+		},
+		{
+			desc: "Empty Backend LB-Drr Sticky",
+			dynamicConfig: func(testServerURL string) *types.Configuration {
+				return &types.Configuration{
+					Frontends: map[string]*types.Frontend{
+						"frontend": {
+							EntryPoints: []string{"http"},
+							Backend:     "backend",
+							Routes:      map[string]types.Route{requestPath: {Rule: "Path:" + requestPath}},
+						},
+					},
+					Backends: map[string]*types.Backend{
+						"backend": {
+							Servers:      map[string]types.Server{},
+							LoadBalancer: &types.LoadBalancer{Method: "Drr", Sticky: true},
+						},
+					},
+				}
+			},
+			wantStatusCode: http.StatusServiceUnavailable,
+		},
+		{
+			desc: "Empty Backend LB-Wrr",
+			dynamicConfig: func(testServerURL string) *types.Configuration {
+				return &types.Configuration{
+					Frontends: map[string]*types.Frontend{
+						"frontend": {
+							EntryPoints: []string{"http"},
+							Backend:     "backend",
+							Routes:      map[string]types.Route{requestPath: {Rule: "Path:" + requestPath}},
+						},
+					},
+					Backends: map[string]*types.Backend{
+						"backend": {
+							Servers:      map[string]types.Server{},
+							LoadBalancer: &types.LoadBalancer{Method: "Wrr"},
+						},
+					},
+				}
+			},
+			wantStatusCode: http.StatusServiceUnavailable,
+		},
+		{
+			desc: "Empty Backend LB-Wrr Sticky",
+			dynamicConfig: func(testServerURL string) *types.Configuration {
+				return &types.Configuration{
+					Frontends: map[string]*types.Frontend{
+						"frontend": {
+							EntryPoints: []string{"http"},
+							Backend:     "backend",
+							Routes:      map[string]types.Route{requestPath: {Rule: "Path:" + requestPath}},
+						},
+					},
+					Backends: map[string]*types.Backend{
+						"backend": {
+							Servers:      map[string]types.Server{},
+							LoadBalancer: &types.LoadBalancer{Method: "Wrr", Sticky: true},
+						},
+					},
+				}
+			},
```

Comment:
Should we add a parameterizable helper function or (maybe even better) apply the builder pattern to construct the frontends and backends for test purposes?

We have tests like the ones in `provider/kubernetes/kubernetes_test.go` which are using inline structs so extensively, it has become fairly hard to understand and extend the tests. I'd like to make sure we're not doing the same for newish tests.